### PR TITLE
Add ability to sort discovered files

### PR DESCRIPTION
**Pull request by stevebauman**
This PR adds the ability to sort the discovered files by all of Symfony's `Finder` sort options.

To sort, we can use the `StructureResolverSort` enum in the `Discover::sortBy()` method:

```php
$sortedFiles = Discover::in('...')->sortBy(StructureResolverSort::NAME)->get();
```

We can also reverse the sorting by passing in `true` into the second parameter:

```php
$reversedFiles = Discover::in('...')->sortBy(StructureResolverSort::NAME, true)->get();
```

I've created this PR as a draft, as I'm unsure if others will agree on the naming of the `StructureResolverSort` enum.

Let me know if you would like anything adjusted. Thanks for your time!

No hard feelings on closure ❤️

### CODE REVIEW

File: src/Enums/StructureResolverSort.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Spatie\StructureDiscoverer\Enums;
+
+use Symfony\Component\Finder\Finder;
+
+enum StructureResolverSort
```
Comment (freekmurze) [inline src/Enums/StructureResolverSort.php:7]: Let's rename this to the more simple `Sort`

File: src/Enums/StructureResolverSort.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    case NAME;
+    case SIZE;
+    case TYPE;
+    case EXTENSION;
+    case CHANGED_TIME;
```
Comment (freekmurze) [inline src/Enums/StructureResolverSort.php:13]: We use `PascalCase` for enums. Could you change that. If you need to have a value with another casing, feel free to add string values to this enum.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Thanks for your work on this!

File: tests/StructureDiscovererTest.php [hunk 3]
```diff
@@ -277,6 +278,54 @@ public function satisfies(DiscoveredStructure $discoveredData): bool
...
+    Sort $sort,
+    array $expected,
+) {
+    $found = Discover::in(__DIR__ . '/Fakes')->classes()->sortBy($sort)->get();
+
+    expect($found)->toEqual($expected);
+})->with(
+    [
+        'name' => [
+            Sort::Name,
```
Comment (stevebauman) [inline tests/StructureDiscovererTest.php:291]: I only included tests for sorting by the file name, as some of the other sort options could change depending on the contents of the file (size), the timestamps (modified, changed, accessed).

If you'd like me to create file stubs for the other testable options like `type` and `size`, let me know! 👍

Comment (freekmurze) [reply tests/StructureDiscovererTest.php:291]: What we have now is fine, thanks!

**Pull request merged**